modules/wallet/wallet_test.go

```diff
@@ -15,6 +16,16 @@ import (
 	"github.com/NebulousLabs/Sia/types"
 )
 
+var (
+	// Sometimes during testing, you get a statement
+	// `if err != expectedErr { return err }`. But if err == nil, then the
+	// calling function will think that the test has passed, even though the
+	// test failed. [This is a developer error, instead return
+	// errors.New("expecting..")] The new type of error catches when the
+	// developer has made a mistake and failed to catch a test suite failure.
+	errTestCompleted = errors.New("test complete")
```

Comment:
https://en.wikipedia.org/wiki/Principle_of_least_astonishment

The implicit understanding is: if this function fails, it will return a non-nil error. So we need to be careful that our functions obey this expected behavior. Calling test functions within test functions is dangerous for this reason. If you're expecting a specific error, you must write:
```go
if err != ErrFoo {
    return fmt.Errorf("expected %v, got %v", ErrFoo, err)
}
```
It's a little annoying, but it's also a fact of language. Trying to hide it with fancy wrappers and special variables is an anti-pattern.

As always, I recommend taking a step back and considering: "have I seen anyone else write code like this?" If you haven't, that should give you pause.